### ISSUE TITLE
Open Scope Type gives surprising parsing error.

### ISSUE THREAD

**Issue by Zimmi48** [coq/coq#14098]
#### Description of the problem

If I mistakenly type `Open Scope Type` instead of `Open Scope type`, I get the surprising parsing error "Syntax error: illegal begin of vernac." with "Open" colored in red. It can take some time to fix it because for that you = need to understand that Coq is not pointing you to the place where the error actually is. Can the parsing error be improved?

#### Coq Version

8.13.1

**Comment by herbelin** [comment]
That's an interesting observation.

This is due to the look-ahead. A sequence `Open`, `Scope`, `IDENT` is expected but not found, so, it backtracked to the beginning of the sequence, finds no other rules starting with `Open` and fails here.

A possible workaround is to replace:
```
   [ [ IDENT "Open"; IDENT "Scope"; sc = IDENT ->  
```
by
```
   [ [ IDENT "Open"; IDENT "Scope"; sc = preident ] ->
```
in `g_vernac.mlg`, so as to deactivate the look-ahead on `IDENT`. And similarly for similar lines.

More generally, should it be considered as an issue with gramlib look-ahead to be addressed there, I don't know. Maybe errors arising from incomplete look-ahead matches with no alternative matching rules should be reported on where the look-ahead made maximal progress rather than at when the look-ahead started.

**Comment by Zimmi48** [comment]
>Maybe errors arising from incomplete look-ahead matches with no alternative matching rules should be reported on where the look-ahead made maximal progress rather than at when the look-ahead started.

Yes that sounds good. But why is there a look ahead here? I don't see any other command starting with `Open`.

**Comment by herbelin** [comment]
There is a (token) look-ahead any time you have a (unique, see #14070) sequence of consecutive tokens.

I otherwise realize that the documentation I wrote in #14075 about the max position reached in the stream is not what actually happens but is the behavior you are expecting [lstream.ml, line 17](https://github.com/coq/coq/pull/14075/commits/43e3561bb701ee6c44c66da31cff428467611943). So, I may try to experiment a change in #14075.
